### ISSUE TITLE
Inquiry on some details of the method.

### ISSUE THREAD

**Issue by leekum2018** [Hzfinfdu/Diffusion-BERT#8]
As said in the second paragraph of Section 4.3, "We attribute the superior performance of DiffusionBERT to its onetime sampling of all tokens". I wonder the meaning of "onetime sampling of all tokens", does it mean generating all the tokens in a sentence at a time. If it does, it seems to conflict with the demonstration in Table 1. Thank you!

**Comment by Hzfinfdu** [comment]
Hi,
 
Yes, we generate all tokens in one diffusion step. We use ddim sampling to predict $x_0$ and get $x_{t-1}$ from the forward process. The demonstration in Table 1 shows the input of BERT at time step $t-1$. 

Besides, the corresponding predicted $x_0$ is composed of less informative tokens when $t$ is large and gradually shows semantic meaning as $t$ goes to 0. That is also the motivation of our spindle noise schedule.

Hope this helps. If you have more questions please feel free to contact with me.

**Comment by leekum2018** [comment]
Thank you for your reply! I have a further question. According to your reply, does it means you model  $p_{\theta}(x_{t-1}|x_t)$ as 
![Screenshot 2022-12-21 at 12 29 17](https://user-images.githubusercontent.com/48008881/208821464-74ddbb8c-ff46-4953-a647-5028fd2b5cf8.png)
And is the second term on the right hand side the output of BERT? Thank you!

**Comment by Hzfinfdu** [comment]
Yes, that's right. DDIM sampling helps to trade off speed and generation quality. And predicting $x_0$ directly is closer to the MLM training objective.

**Comment by leekum2018** [comment]
Hi, I have another question. In eq. 9, how to compute  $H(x_{0}^{i})$, in other words, what is the distribution of  $x_{0}^{I}$ for calculating $H(x_{0}^{i})$. Because I have a hard time understanding  why the following equation holds. 
![Screenshot 2022-12-27 at 13 33 19](https://user-images.githubusercontent.com/48008881/209616290-9dcc9837-25d7-403b-9e76-82d2e12b7a91.png)
Thank you!

**Comment by Hzfinfdu** [comment]
Hi, 

In fact, $H(x_0^i)$ can be calculated in many ways. We calculate the entropy of each token by the negative logarithm of its frequency in the tokenized training corpus.

Since a masked token loses all its information, the expected information loss of the i-th token at $t$ is $\overline{\alpha}_t^iH(\textbf{x_o^i})$. We get Eq. 9 by taking the sum over the sequence.

Hope this helps.

**Comment by leekum2018** [comment]
Assume a training corpus with only one sentence $x$ as "A A B", is it that $H(x_0^0)=H(x_0^1)=-(2/3) log (2/3)$ or $H(x_0^0)=H(x_0^1)=-(2/3) log (2/3)-(1/3) log (1/3)$?
Thank you!

**Comment by leekum2018** [comment]
For the following formula _Structured Denoising Diffusion Models in Discrete State-Spaces_, why the RHS is proportional to RHS? Could you please give me some hints? I have a hard time deriving this. Please.
![image](https://user-images.githubusercontent.com/48008881/211550932-b896f5cd-48da-4104-bd6c-5841b8f0c54f.png)

**Comment by Siddharth-Shrivastava7** [comment]
Hi @leekum2018, 

you can refer this : https://openreview.net/forum?id=h7-XixPCAL&noteId=xm7onR_Sg0L 

Hope it helps!
